dvc/output/base.py

```diff
@@ -297,8 +297,9 @@ def checkout(self, force=False, progress_callback=None, tag=None):
             progress_callback=progress_callback,
         )
 
-    def remove(self, ignore_remove=False):
-        self.remote.remove(self.path_info)
+    def remove(self, ignore_remove=False, force=True):
+        self.remote.safe_remove(self.path_info, force=force)
```

Comment:
But `self.remote` might not be the same as `self.cache` and `self.cache` is the one that actually has cache and can check if cache for that path_info is present. I'm pretty sure you've had a `if self.use_cache` here previously, no?

Reply:
This is solved on this one:
```diff
    def already_cached(self, path_info):
        assert path_info.scheme in ["", "local"]

        current_md5 = self.get_checksum(path_info)

        if not current_md5:
            return False

-        return not self.changed_cache(current_md5)
+        return not self.cache.changed_cache(current_md5)
```

It doesn't matter if you call it with `remote` or `cache` (It thought this was a better interface / less error prone) :sweat_smile:  at the end, it will call the `cache`, the tricky part is to get that you can call `cache.cache.cache....` multiple times and it will return the same instance

Reply:
The `if self.use_cache` is on `changed_cache` already :thinking: